apitools/base/py/transfer.py

```diff
@@ -438,7 +440,7 @@ def GetRange(self, start, end=None, additional_headers=None,
             progress_end_normalized = True
         else:
             progress = start
-        while not progress_end_normalized or progress < end:
+        while not progress_end_normalized or progress < end + 1:
```

Comment:
This is a bug (can repro with gsutil cat -r 1-1 gs://bucket/object), but GetRange with end < size is only used for the cat command with the range flag.  Otherwise, we're always downloading the object all the way to the end, so we won't hit this.